match_tpl.h

```diff
@@ -89,50 +117,80 @@ int32_t LONGEST_MATCH(deflate_state *const s, Pos cur_match) {
          * is limited to the lookahead, so the output of deflate is not
          * affected by the uninitialized values.
          */
-        cont = 1;
-        do {
-            match = window + cur_match;
-            if (LIKELY(*(bestcmp_t *)(match+best_len-1) != scan_end ||
-                       *(bestcmp_t *)(match) != scan_start)) {
-                if ((cur_match = prev[cur_match & wmask]) > limit && --chain_length != 0) {
-                    continue;
-                }
-                cont = 0;
+#ifdef UNALIGNED_OK
+        if (best_len < sizeof(uint32_t)) {
```

Comment:
Here we check for smaller than sizeof(unit32_t)

Reply:
We do this check for best_len < 4 so we know to match using 2 byte match at beginning and end. This uses the `best_len-1` offset. 1 past byte plus 1 new byte.